File: MultiplyLists/ml.py
# Multiply Lists solution in Python 2 for CodeEval.com by Steven A. Dunn

import sys

for line in open(sys.argv[1], 'r'):
  line = line.rstrip('\n').split(" | ")
  line = [i.split(' ') for i in line] 
  line = [map(int, i) for i in line] 

  result = map(lambda x,y: x * y, line[0], line[1])
  
  for n in result:
    print n,
  print

File: MultiplyLists/ml.rb
# multiply lists soln in rb for codeeval by steven a dunn

File.open(ARGV[0], 'r').each_line do |line|
    line = line.chomp.split(" | ")
    line1 = line[0].split(' ').map(&:to_i)
    line2 = line[1].split(' ').map(&:to_i)

    line1.zip(line2).each do |x, y|
        print "#{x * y} "
    end
    puts
end